tests/qxmpppushenableiq/tst_qxmpppushenableiq.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (C) 2008-2020 The QXmpp developers
+ *
+ * Author:
+ *  Jonah Brüchert
+ *
+ * Source:
+ *  https://github.com/qxmpp-project/qxmpp
+ *
+ * This file is a part of QXmpp library.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ */
+
+#include <QObject>
+#include "util.h"
+#include <QXmppPushEnableIq.h>
+#include <QXmppDataForm.h>
+
+class tst_QXmppPushEnableIq : public QObject
+{
+    Q_OBJECT
+
+private slots:
+    void testPushEnable();
+    void testPushDisable();
+    void testXmlNs();
+    void testDataForm();
+    void testIsEnableIq();
+};
+
+void tst_QXmppPushEnableIq::testPushEnable()
+{
+    const QByteArray xml(
+        R"(<iq id="x42" type="set">)"
+        R"(<enable xmlns="urn:xmpp:push:0" jid="push-5.client.example" node="yxs32uqsflafdk3iuqo"/>)"
+        "</iq>");
+
+    QXmppPushEnableIq iq;
+    parsePacket(iq, xml);
+    QCOMPARE(iq.mode(), QXmppPushEnableIq::Enable);
+    QCOMPARE(iq.jid(), "push-5.client.example");
+    QCOMPARE(iq.node(), "yxs32uqsflafdk3iuqo");
+
```

Comment:
Please also serialize the `iq` to ensure parsing + serializing returns exactly the same XML again.